### PR TITLE
Have different cache directories for different servers

### PR DESCRIPTION
**Pull request by mfeurer**

### CODE REVIEW

**Comment by mfeurer** [comment]
This now also partially solves #396. However, not as nicely as expected.

File: tests/test_setups/test_setup_functions.py [hunk 1]
```diff
@@ -111,7 +111,7 @@ def test_existing_setup_exists_3(self):
 
     def test_get_setup(self):
         # no setups in default test server
-        openml.config.server = 'https://www.openml.org/api/v1/xml/'
+        openml.config.set_server_url('https://www.openml.org/api/v1/xml/')
```
Comment (janvanrijn) [inline tests/test_setups/test_setup_functions.py]: why is this still a fn?

File: openml/testing.py [hunk 2]
```diff
@@ -66,8 +68,14 @@ def setUp(self):
 
     def tearDown(self):
         os.chdir(self.cwd)
-        shutil.rmtree(self.workdir)
+        for i in range(10):
```
Comment (janvanrijn) [inline openml/testing.py]: make 10 a variable?

**Comment by ArlindKadra** [comment]
@mfeurer @janvanrijn

**Comment by mfeurer** [comment]
Thanks for the fix @ArlindKadra. Could you please describe which change was actually necessary to fix it? Also, could you please explain the reasoning for changing the `static_cache_dir` variable?

**Pull request merged**